src/Plugin/Validation/Constraint/InstagramEmbedCodeConstraintValidator.php

```diff
@@ -19,14 +19,14 @@ class InstagramEmbedCodeConstraintValidator extends ConstraintValidator {
   /**
    * {@inheritdoc}
    */
-  public function validate($entity, Constraint $constraint) {
-    if (!isset($entity)) {
+  public function validate($value, Constraint $constraint) {
+    if (!isset($value)) {
       return;
     }
 
     $matches = [];
     foreach (Instagram::$validationRegexp as $pattern => $key) {
-      if (preg_match($pattern, $entity->value, $item_matches)) {
+      if (preg_match($pattern, $value, $item_matches)) {
```

Comment:
I have tried with this code and at least for the link type it works. Haven t check for the other types yet...

```php
/**
 * Validates the InstagramEmbedCode constraint.
 */
class InstagramEmbedCodeConstraintValidator extends ConstraintValidator {

  /**
   * {@inheritdoc}
   */
  public function validate($entity, Constraint $constraint) {
    if (!isset($entity)) {
      return;
    }

    $matches = [];
    foreach (Instagram::$validationRegexp as $pattern => $key) {
      if (preg_match($pattern, $this->getEmbedCode($entity), $item_matches)) {
        $matches[] = $item_matches;
      }
    }

    if (empty($matches)) {
      $this->context->addViolation($constraint->message);
    }
  }

  /**
   * Extracts the embed code from a field item.
   *
   * @param mixed $value
   *   The value that should be validated.
   *
   * @return string|null
   *   The embed code, or NULL if the field type is not supported.
   */
  protected function getEmbedCode($entity) {

    switch ($entity->getFieldDefinition()->getType()) {
      case 'link':
        return $entity->uri;
      case 'string':
      case 'string_long':
        return $entity->value;
      default:
        break;
    }
  }

}
```